cl/opinion_page/tests.py

```diff
@@ -1523,3 +1533,145 @@ async def test_block_cluster_and_docket_via_ajax_view(self) -> None:
+        self.mocked_docket_entries = [de1, de1_2, de2, de3]
+        self.mocked_extra_docket_entries = [de4]
+
+        request_factory = RequestFactory()
+        self.request = request_factory.get("/mock-url/")
+        self.user = UserFactory.create(
+            username="learned",
+            email="learnedhand@scotus.gov",
+        )
+        self.request.auser = AsyncMock(return_value=self.user)
+
+    def tearDown(self):
+        # Clear all test data
+        Docket.objects.all().delete()
+        DocketEntry.objects.all().delete()
+        RECAPDocument.objects.all().delete()
+        User.objects.all().delete()
+
+    def test_fetch_docket_entries(self) -> None:
+        """Verify that fetch entries function returns right docket_entries"""
+        res = asyncio.run(fetch_docket_entries(self.mocked_docket))
+        self.assertEqual(len(res), len(self.mocked_docket_entries))
+        self.assertIn(self.mocked_docket_entries[0], res)
+        self.assertNotIn(self.mocked_extra_docket_entries[0], res)
+
+    def test_generate_docket_entries_csv_data(self) -> None:
+        """Verify str with csv data is created. Check column and data entry"""
+        res = generate_docket_entries_csv_data(self.mocked_docket_entries)
+        res_lines = res.split("\r\n")
+        res_line_data = res_lines[1].split(",")
+        self.assertEqual(res[:16], '"docketentry_id"')
+        self.assertEqual(res_line_data[1], '"506581111"')
+
+    def test_view_download_docket_entries_csv(self) -> None:
+        """Test download_docket_entries_csv returns csv content"""
+        with mock.patch(
+            "cl.opinion_page.utils.generate_docket_entries_csv_data"
+        ) as mock_download_function:
+            with mock.patch(
+                "cl.opinion_page.utils.core_docket_data"
+            ) as mock_core_docket_data:
+                with mock.patch(
+                    "cl.opinion_page.utils.user_has_alert"
+                ) as mock_user_has_alert:
+                    mock_download_function.return_value = (
+                        '"col1","col2","col3"\r\n"value1","value2","value3"'
+                    )
+                    mock_user_has_alert.return_value = False
+                    mock_core_docket_data.return_value = (
+                        self.mocked_docket,
+                        {
+                            "docket": self.mocked_docket,
+                            "title": "title",
+                            "note_form": "note_form",
+                            "has_alert": mock_user_has_alert.return_value,
+                            "timezone": "EST",
+                            "private": True,
+                        },
+                    )
+
+                    response = async_to_sync(download_docket_entries_csv)(
+                        self.request, self.mocked_docket.id
+                    )
+                    self.assertEqual(response["Content-Type"], "text/csv")
```

Comment:
While mocking is a great testing strategy, nested context managers can sometimes hinder code clarity. In this instance, using the `@mock.patch` decorator provides a more concise and readable approach to achieving the same goals.

```python
    @mock.patch("cl.opinion_page.utils.user_has_alert")
    @mock.patch("cl.opinion_page.utils.core_docket_data")
    @mock.patch("cl.opinion_page.utils.generate_docket_entries_csv_data")
    def test_view_download_docket_entries_csv(
        self,
        mock_download_function,
        mock_core_docket_data,
        mock_user_has_alert,
    ) -> None:
        """Test download_docket_entries_csv returns csv content"""
        mock_download_function.return_value = (
            '"col1","col2","col3"\r\n"value1","value2","value3"'
        )
        mock_user_has_alert.return_value = False
        mock_core_docket_data.return_value = (
            self.mocked_docket,
            {
                "docket": self.mocked_docket,
                "title": "title",
                "note_form": "note_form",
                "has_alert": mock_user_has_alert.return_value,
                "timezone": "EST",
                "private": True,
            },
        )

        response = async_to_sync(download_docket_entries_csv)(
            self.request, self.mocked_docket.id
        )
        self.assertEqual(response["Content-Type"], "text/csv")
```

Reply:
It makes sense. Done as suggested